### DISCUSSION TITLE
Link: wait for page unmount before scrolling to the top

### DISCUSSION THREAD

**Discussion by Bleuzen** [vercel/next.js#23944] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

When using page transitions, the page scrolls to the top when clicking on a Link while the exit animation of the old page is still running.

## Describe the solution you'd like

Let Link wait for the old component to fully unmount before scrolling to the top.

## Describe alternatives you've considered

When using framer-motion, an alternative is to use the workaround used in this example:
https://github.com/vercel/next.js/tree/canary/examples/with-framer-motion

1) Set `scroll={false}` on all Links
2) Scroll to top in framer-motions `onExitComplete` event

But setting `scroll={false}` on every Link is really annoying and easy to forget.

EDIT:
Also when using the browsers back / forward buttons, that pages also scroll to the top. Wonder if there is a way to prevent this, because I can't set `scroll={false}` in this case, because it isn't triggered by Links. So browser navigation should also wait for page unmount before scrolling to the top.

**Comment by vladfulgeanu** [comment]
Currently experiencing this in 9.5.1.
It kind of ruins the whole experience of fast page transitions with different scrolling offsets.
Is the bug confirmed @Timer @timneutkens?
